### PR TITLE
health: Add and Start a Health Probe Server

### PR DESCRIPTION
**Pull request by delqn**
This PR implements Liveness and Readiness probes for the AGIC container.

The port on which the Health Probe Server runs is defaulted to `8123`.
This is configurable via the `HEALTH_PROBE_SERVICE_PORT` environment variable / `kubernetes.healthProbeServicePort` Helm template variable.

This is laying the foundation for these probes.
Liveness at this state is true when the server is running
Readiness is when the K8s cache has synced.

The URLs for the probes are:
  - `/health/ready`
  - `/health/alive`

----
From `kubectl describe pod X` after `helm install`:
```
Containers:
  ingress-azure:
    Container ID:   docker://7b4680ccef46f4123355d26d1b7c7090c19a350121e96cc9740ca1237b69ae1b
    Image:          xxx.azurecr.io/kubernetes-ingress:0.1x
    Image ID:       docker-pullable://xxx.azurecr.io/kubernetes-ingress@sha256:911df860fe58c1f44809d6035282d8f3f4dd3b88521b894dd453a1832d234da4
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Wed, 31 Jul 2019 11:40:07 -0700
    Ready:          True
    Restart Count:  0
    Liveness:       http-get http://:8123/health/alive delay=15s timeout=1s period=20s #success=1 #failure=3
    Readiness:      http-get http://:8123/health/ready delay=5s timeout=1s period=10s #success=1 #failure=3
```

---

From the pod's logs:
```
I0731 18:40:08.058459       1 main.go:142] Starting Health Probe Server on :8123
```

---

Quick test: `kubectl port-forward ingress-azure-5875459fb6-bm7mr 8123:8123`
```bash
curl -I localhost:8123/health/nothing

HTTP/1.1 404 Not Found
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Wed, 31 Jul 2019 18:46:27 GMT
Content-Length: 19
```

```bash
curl -I localhost:8123/health/ready

HTTP/1.1 200 OK
Date: Wed, 31 Jul 2019 18:46:27 GMT
```

```bash
curl -I localhost:8123/health/alive

HTTP/1.1 200 OK
Date: Wed, 31 Jul 2019 18:46:27 GMT
```

### CODE REVIEW

File: helm/ingress-azure/templates/deployment.yaml [hunk 1]
```diff
@@ -30,6 +30,34 @@ spec:
       - name: {{ .Chart.Name }}
         image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        readinessProbe:
+          httpGet:
+            path: /health/ready
+        {{- if .Values.kubernetes }}
+        {{- if .Values.kubernetes.healthProbeServicePort }}
+            port: {{ .Values.kubernetes.healthProbeServicePort }}
```
Comment (akshaysngupta) [inline helm/ingress-azure/templates/deployment.yaml]: give `Values.kubernetes.healthProbeServicePort` a default value in values.yml ?

File: helm/ingress-azure/templates/deployment.yaml [hunk 2]
```diff
@@ -30,6 +30,34 @@ spec:
...
+        {{ else }}
+            port: 8123
+        {{- end}}
+        {{ else }}
+            port: 8123
```
Comment (akshaysngupta) [inline helm/ingress-azure/templates/deployment.yaml]: is this twice ?

File: helm/ingress-azure/templates/deployment.yaml [hunk 3]
```diff
@@ -30,6 +30,34 @@ spec:
...
+          httpGet:
+            path: /health/alive
+        {{- if .Values.kubernetes }}
+        {{- if .Values.kubernetes.healthProbeServicePort }}
+            port: {{ .Values.kubernetes.healthProbeServicePort }}
+        {{ else }}
+            port: 8123
+        {{- end}}
+        {{ else }}
+            port: 8123
```
Comment (akshaysngupta) [inline helm/ingress-azure/templates/deployment.yaml]: same for this.

File: helm/ingress-azure/templates/deployment.yaml (hunk 2 above)
Comment (delqn) [reply helm/ingress-azure/templates/deployment.yaml]: Added a default value for `healthProbeServicePort`  in `values.yaml`
This was if/then/elsed twice as a workaround Go's templates (or perhaps my lack of understanding these)

File: helm/ingress-azure/templates/deployment.yaml (hunk 1 above)
Comment (delqn) [reply helm/ingress-azure/templates/deployment.yaml]: Good idea! Thanks!

File: helm/ingress-azure/templates/deployment.yaml (hunk 3 above)
Comment (delqn) [reply helm/ingress-azure/templates/deployment.yaml]: Added a default value for `healthProbeServicePort`  in `values.yaml`

**Pull request merged**